moda-contracts/src/Releases.sol

```diff
@@ -0,0 +1,204 @@
+    address _catalog;
+    address _splitsFactory;
+
+    string public name;
+    string public symbol;
+
+    uint256 public numberOfReleases;
+
+    bool _isOpen;
+
+    mapping(uint256 => string) _uris;
+
+    /// @dev users with a Releases admin role can create releases with a curated contract
+    bytes32 public constant RELEASE_ADMIN_ROLE = keccak256("RELEASE_ADMIN_ROLE");
+
+    // Errors
+    error CannotBeZeroAddress();
+    error InvalidAmount();
+    error FieldCannotBeEmpty(string field);
+    error CallerIsNotReleaseAdmin();
+    error InvalidTokenId();
+    error CallerDoesNotHaveTrackAccess();
+
+    /**
+     * @dev Constructor
+     * @param isInitializerDisabled - Whether the initializer is disabled or not
+     */
+    constructor(bool isInitializerDisabled) {
+        if (isInitializerDisabled) {
+            _disableInitializers();
+        }
+    }
+
+    // External Functions
+
+    /**
+     * @inheritdoc IReleases
+     */
+    function initialize(
+        address admin,
+        address[] calldata releaseAdmins,
+        string calldata name_,
+        string calldata symbol_,
+        address catalog,
+        address splitsFactory
+    ) external initializer {
+        __ERC1155_init("");
+        if (admin == address(0)) revert CannotBeZeroAddress();
+        if (bytes(name_).length == 0) revert FieldCannotBeEmpty("name");
+        if (bytes(symbol_).length == 0) revert FieldCannotBeEmpty("symbol");
+        _grantRole(DEFAULT_ADMIN_ROLE, admin);
+        for (uint8 i = 0; i < releaseAdmins.length; i++) {
+            _grantRole(RELEASE_ADMIN_ROLE, releaseAdmins[i]);
+        }
+        name = name_;
+        symbol = symbol_;
+        _catalog = catalog;
+        _splitsFactory = splitsFactory;
+    }
+
+    /**
+     * @inheritdoc IReleases
+     */
+    function create(
+        address receiver,
+        uint96 royaltyAmount,
+        address beneficiary,
+        /// will be an array of addresses to pass to splits factory
+        string calldata uri_,
+        uint256 amount,
+        string[] calldata trackIds
+    ) external nonReentrant {
+        if (_isOpen) {
+            _requireCallerHasTrackAccess(trackIds);
+        } else if (!hasRole(RELEASE_ADMIN_ROLE, msg.sender)) {
+            revert CallerIsNotReleaseAdmin();
+        }
+        if (royaltyAmount > 2_000) revert InvalidAmount();
+
+        numberOfReleases++;
+        _uris[numberOfReleases] = uri_;
+        /// Create splits address here
+        IReleaseRegistration(_catalog).registerRelease(trackIds, uri_, numberOfReleases);
+        _mint(receiver, numberOfReleases, amount, "");
+        _setTokenRoyalty(numberOfReleases, beneficiary, royaltyAmount);
+
+        emit ReleaseCreated(numberOfReleases);
+    }
+
+    /**
+     * @inheritdoc IReleases
+     */
+    function withdrawRelease(
+        address receiver,
+        uint256 tokenId,
+        uint256 amount
+    ) external onlyRole(RELEASE_ADMIN_ROLE) nonReentrant {
```

Comment:
How does this work if the contract is "open". Let's say I create a release on an open contract. Will I be able to withdraw that? If I created it shouldn't I own it?